databases/src/connect/ConnectDatabaseAccountPromptStep.ts

```diff
@@ -0,0 +1,158 @@
+                    databaseTreeItems.push(databaseItem);
+
+                } else {
+                    const masterKeyResult = (await cosmosClient.databaseAccounts.listKeys(azureUtils.azureUtils.getResourceGroupFromId(accountId), accountName))._response.parsedBody;
+
+                    const endpoint = nonNullProp(account, 'documentEndpoint');
+                    const masterKey = nonNullProp(masterKeyResult, 'primaryMasterKey');
+                    const databaseItem: DBTreeItem = { hostName: accountName, port: '5432', connectionString, azureData: { accountId: accountId, accountName: accountName, resourceGroup: azureUtils.azureUtils.getResourceGroupFromId(accountId), accountKind: getExperienceLabel(account) }, docDBData: { masterKey, documentEndpoint: endpoint } };
+                    databaseTreeItems.push(databaseItem);
+                }
+
+            }
+
+        }
+
+        return databaseTreeItems;
+    }
+
+    public async prompt(context: IConnectDBWizardContext): Promise<void> {
+        const options: IAzureQuickPickOptions = { placeHolder: 'Add Database Connection.' };
+        const input: IAzureQuickPickItem<DBTreeItem | undefined> = await context.ui.showQuickPick(this.getQuickPickItems(context), options);
+        if (input.label === skipForNowLabel) {
+            context.addDBConnectionSkip = true;
+        } else if (input.label.includes('Create')) {
+            context.createDBAccount = true;
+        } else if (input.data) {
+            context.databaseConnectionTreeItem = input.data;
+        }
+    }
+
+    public shouldPrompt(context: IConnectDBWizardContext): boolean {
+        return !context.databaseAccountName;
+    }
+
+    public async getSubWizard(context: IConnectDBWizardContext): Promise<IWizardOptions<IConnectDBWizardContext> | undefined> {
+        if (context.createDBAccount) {
+            const promptSteps: AzureWizardPromptStep<ILocationWizardContext>[] = [new AzureDBAPIStep(), new ResourceGroupListStep()];
+
+            LocationListStep.addStep(context, promptSteps);
+            return {
+                promptSteps: promptSteps,
+                executeSteps: []
+            }
+        }
+        return undefined;
+    }
+
+    private async getQuickPickItems(context: IConnectDBWizardContext): Promise<IAzureQuickPickItem<DBTreeItem | undefined>[]> {
+        const picks: IAzureQuickPickItem<DBTreeItem | undefined>[] = !this._suppressCreate ? [{
+            label: localize('newDBConnection', '$(plus) Create new Azure Database'),
+            data: undefined
+        }] : [];
+        picks.push({
+            label: localize('skipForNow', skipForNowLabel),
+            data: undefined
+        });
+
+        let components: DBTreeItem[] = [];
+        const api: API = nonNullProp(context, 'defaultExperience').api;
+        switch (api) {
+            case API.Core:
```

Comment:
This approach is totally valid, but you can also do something like this to remove a little bit of duplicate code:
```typescript
    case API.Core:
    case API.MongoDB:
        components.push(...await ConnectDatabaseAccountPromptStep.getCosmosDatabases(context, api));
        break;
```

And the same below for `API.PostgresSingle` and `default`.